src/lib/Field.tsx

```diff
@@ -25,11 +27,27 @@ export const Field = <FP extends FieldPath.FieldPath<any, any> = string>({
   const isDirty = useSelector(() => toolkit.isFieldDirty(fieldPath));
   const [isValid, setIsValid] = useState(true);
 
-  const refs = useRef({ validate });
+  const refs = useRef({ validate, promiseCounter: 0 });
   Object.assign(refs.current, { validate });
 
   useEffect(() => {
-    setIsValid(refs.current.validate?.(value) ?? true);
+    const promiseId = ++refs.current.promiseCounter;
+    let result = refs.current.validate?.(value) ?? true;
+    if (result instanceof Promise) {
+      result.then((valid) => {
+        if (promiseId !== refs.current.promiseCounter) {
+          return;
+        }
+        setIsValid(valid);
+        refs.current.promiseCounter = 0;
+      });
+    } else {
+      if (promiseId !== refs.current.promiseCounter) {
```

Comment:
Looks like this condition is unnecessary. Nothing async happend, this always returns false.